client/fidelity/compliance-review/compliance-review.js

```diff
@@ -0,0 +1,78 @@
+ComplianceReviewCtrl.$inject = ['$location', 'config', 'moment'];
+export function ComplianceReviewCtrl($location, config, moment) {
+    var VIEW_DATE_FORMAT = config.view.dateformat;
+
+    const SUPERDESK = 'local';
+
+    let deadline = $location.search()['deadline'];
+
+    if(deadline !== 'month' && deadline !== 'week') {
+        $location.search('deadline', 'month');
+        deadline = $location.search()['deadline'];
+    }
+
+    // methods for view
+
+    this.isActive = (deadline) => $location.search()['deadline'] === deadline;
+
+    this.setDeadline = (deadline) => {
+        $location.search('deadline', deadline);
+    };
+
+    // methods for parent directive
+
+    this.repo = {
+        published: true,
+        search: SUPERDESK,
+    };
+
+    this.getSearch = () => {
+        let deadline = $location.search()['deadline'];
+
+        if(deadline !== 'month' && deadline !== 'week') {
+            // in case page is changed, but it didn't finish removing old references yet
+            return {};
+        }
+
+        const now = new Date();
+
+        let dateFrom;
+        if(deadline === 'week') {
+            dateFrom = new Date(now.getFullYear() - 1, now.getMonth(), now.getDate() - 14);
```

Comment:
this could be handled via elastic date math `now-1y-2w/d`

Reply:
Yeah, I know, but the search would need to be refactored to make use of it. I'd like to first finalize the feature and see if I'd still use the search view by then.